tests/shared/core/test_domain.py

```diff
@@ -1750,3 +1750,127 @@ def test_slot_mappings_ignored_intents_during_active_loop():
     assert (
         SlotMapping.intent_is_desired(mappings_for_cuisine[0], tracker, domain) is False
     )
+
+
+def test_domain_slots_for_entities_with_mapping_conditions_no_slot_set():
+    domain = Domain.from_yaml(
+        textwrap.dedent(
+            """
+            entities:
+            - city
+            slots:
+              location:
+                type: text
+                influence_conversation: false
+                mappings:
+                - type: from_entity
+                  entity: city
+                  conditions:
+                  - active_loop: booking_form
+            forms:
+              booking_form:
+                required_slots:
+                  - location
+            """
+        )
+    )
+    events = domain.slots_for_entities([{"entity": "city", "value": "Berlin"}])
+    assert len(events) == 0
+
+
+def test_domain_slots_for_entities_sets_valid_slot():
+    domain = Domain.from_yaml(
+        textwrap.dedent(
+            """
+            entities:
+            - city
+            slots:
+              location:
+                type: text
+                influence_conversation: false
+                mappings:
+                - type: from_entity
+                  entity: city
+            """
+        )
+    )
+    events = domain.slots_for_entities([{"entity": "city", "value": "Berlin"}])
+    assert events == [SlotSet("location", "Berlin")]
+
+
+def test_invalid_domain_from_trigger_intent_mapping_slot_not_in_forms():
+
+    with pytest.raises(InvalidDomain) as e:
+        Domain.from_yaml(
+            textwrap.dedent(
+                """
+            intents:
+            - activate_booking
+            entities:
+            - city
+            slots:
+              location:
+                type: text
+                influence_conversation: false
+                mappings:
+                - type: from_entity
+                  entity: city
+                  conditions:
+                  - active_loop: booking_form
+              started_booking_form:
+                type: bool
+                influence_conversation: false
+                mappings:
+                - type: from_trigger_intent
+                  intent: activate_booking
+                  value: true
+            forms:
+              booking_form:
+                required_slots:
+                  - location
+            """
+            )
+        )
+
+    assert (
+        "Slot 'started_booking_form' has a from_trigger_intent mapping, "
+        "but it's not listed in any form 'required_slots'." in str(e)
+    )
```

Comment:
you can use the `match` operator in `pytest.raises(InvalidDomain, match="...")` instead